src/core/gui/sidebar/previews/base/SidebarPreviewBaseEntry.cpp

```diff
@@ -67,6 +68,12 @@ auto SidebarPreviewBaseEntry::drawCallback(GtkWidget* widget, cairo_t* cr, Sideb
     return true;
 }
 
+auto SidebarPreviewBaseEntry::sizeChangedCallback(GtkWidget* widget, GtkAllocation* allocation,
+                                                  SidebarPreviewBaseEntry* preview) -> gboolean {
+    preview->repaint();
+    return true;
+}
```

Comment:
This may be a bit too expensive: during a "border drag" sequence, I assume this gets called for every input event, so potentially every ms or so (depending on the input device).
It may be cheaper to test in `paint()` (which is at worst called every ~20ms) whether the size of the buffer matches the size of the widget:

- if the buffer is too big, blit it to a smaller surface and swap them.
- if the buffer is too small, paint it with a rescaling and call for an asynchronous rerendering via `repaint()`
- if the sizes match, just blit it.

Reply:
I will check this out, haven't had the time so far. It's not relevant for "border drag" sequences any more (where a new size was only requested if the border was moved by at least 10 pixels or so), but still relevant for scrolling.